cli/src/main/java/io/github/bmarwell/jdtfmt/JdtFmt.java

```diff
@@ -142,6 +137,45 @@ public static void main(String[] args) {
         }
     }
 
+    public void init() {
+        this.ansiMode = this.noColor ? CommandLine.Help.Ansi.OFF : CommandLine.Help.Ansi.AUTO;
+        this.fmtMode = getFormatterMode();
+        this.formatProcessor = new FormatProcessor(this.fmtMode, new DiffOptions(this.unified[0]));
+    }
+
+    private FormatterMode getFormatterMode() {
+        if (this.write) {
+            // write implies list
+            return FormatterMode.WRITE;
+        }
+
+        if (this.listOnly) {
+            if (this.reportAll) {
+                return FormatterMode.LIST_ALL;
+            } else {
+                return FormatterMode.LIST_FIRST;
+            }
+        }
+
+        if (this.reportAsDiff) {
+            if (this.unified[0] >= 0) {
+                if (this.reportAll) {
+                    return FormatterMode.DIFF_ALL_UNIFIED;
+                } else {
+                    return FormatterMode.DIFF_FIRST_UNIFIED;
+                }
+            } else {
+                if (this.reportAll) {
+                    return FormatterMode.DIFF_ALL_NORMAL;
+                } else {
+                    return FormatterMode.DIFF_FIRST_NORMAL;
+                }
+            }
+        }
+
+        return FormatterMode.PRINT;
```

Comment:
I think `FormatterMode.PRINT` should imply 'quiet' mode, as in, it should print the formatted file, and _only_ the formatted file. That will make it work more easily with other tools (and not just Vim :smile:)
Maybe for some of the diff modes too, I'm not sure...

Reply:
Ah no, not yet